src/transport/raw/MessageHeader.h

```diff
@@ -117,23 +117,29 @@ class PacketHeader
 
     const Header::Flags & GetFlags() const { return mFlags; }
 
+    /** Sets the given flag bit(s). */
+    inline PacketHeader & AddFlag(uint16_t flag)
```

Comment:
I wonder if this function should take a struct Flags?  this would change the definition of the constants to be:

`static constexpr Flags kSecureSessionControlMessage = { .value = 0x0400 };`

for stricter typing

Reply:
I would do this as a future pass - for now my goal was to remove the repeated cast to uint16_t and have it only once. I would love more typesafety as well. We could make flags enum classes and have total safety, however that is probably a larger patch.

Reply:
Updated to be typesafe (separate class, larger PR) after Grant asked me to centralize bit flag handling.

Moved 'FlagUtils.hpp' to 'BitFlags.h' in support and created a typesafe flags class. Old flag classes do not seem typesafe so I marked them in doccomment as deprecated, however I did not remove them as fixing enums within existing code is a larger task.